### PR TITLE
Scala: simplify the sidecar image

### PR DESCRIPTION
**Pull request by sideeffffect**
Install only
 * JRE
 * wrapper script for sbt and mill
 * coursier
 * ammonite

Anything else, the VS Code Metals extension installs itself

closes https://github.com/che-dockerfiles/che-sidecar-scala/pull/9
closes https://github.com/che-dockerfiles/che-sidecar-scala/pull/11

Signed-off-by: sideeffffect <ondra.pelech@gmail.com>

/cc @ericwill

### CODE REVIEW

**Comment by ericwill** [comment]
We still don't have full automation (it's coming soon!) so we need to still provide 2 PR's: one for the meta.yaml update and one for the sidecar. I would propose you leave this PR for the sidecar update only, and update the meta.yaml in #668

**Comment by sideeffffect** [comment]
done :+1:

**Comment by ericwill** [comment]
I think you're missing some changes fromhttps://github.com/eclipse/che-plugin-registry/pull/682, can you rebase on master?

**Comment by ericwill** [comment]
Build check failed: https://github.com/eclipse/che-plugin-registry/pull/684/checks?check_run_id=1382250754

File: sidecars/scala/Dockerfile [hunk 1]
```diff
@@ -24,27 +24,18 @@ RUN apk --no-cache add openjdk11-jre-headless --repository=http://dl-cdn.alpinel
...
+    curl -Ls https://raw.githubusercontent.com/lefou/millw/main/millw > /usr/local/bin/mill && \
+    chmod 0755 /usr/local/bin/mill && \
+    curl -Ls https://git.io/coursier-cli > /usr/local/bin/cs && \
+    chmod 0755 /usr/local/bin/cs && \
+    (echo "#!/usr/bin/env sh" && curl -L https://github.com/lihaoyi/Ammonite/releases/download/2.2.0/2.12-2.2.0) > /usr/local/bin/amm && \
+    chmod 0755 /usr/local/bin/amm
 
 ADD etc/entrypoint.sh /entrypoint.sh
 
-RUN for f in "/projects" "$HOME/.cache" "$HOME/.ivy2" "$HOME/.sbt"; do \
```
Comment (gattytto) [inline sidecars/scala/Dockerfile:44]: this is the fix for the cache folders ownership error don't remove them

File: sidecars/scala/Dockerfile [hunk 2]
```diff
@@ -24,27 +24,18 @@ RUN apk --no-cache add openjdk11-jre-headless --repository=http://dl-cdn.alpinel
 
 WORKDIR /projects
 
-ENV JAVA_HOME=/usr/lib/jvm/default-jvm/ \
-    SBT_VERSION="1.3.10" \
-    METALS_VERSION="0.9.0+10-f3f3e535-SNAPSHOT"
-
-RUN mkdir -p $HOME/.cache && ln -s $HOME/.cache /root/.cache && \
```
Comment (gattytto) [inline sidecars/scala/Dockerfile:31]: let this back in and the ownership change fix will work, then this folders will be available to the extension when it needs them

**Comment by sideeffffect** [comment]
Fixed :heavy_check_mark:

**Comment by gattytto** [comment]
@sideeffffect if you like to, we can also chat in mattermost.eclipse.org channel #Eclipse Che

**Comment by sideeffffect** [comment]
@ericwill I think we can merge this :pray:

**Review by ericwill — APPROVED** [review]

**Comment by ericwill** [comment]
> @ericwill I think we can merge this pray

Done! The resulting image will be tagged `scala-<commit_id>` in https://quay.io/repository/eclipse/che-plugin-sidecar?tag=latest&tab=tags

**Pull request merged**